src/app/CommandHandler.cpp

```diff
@@ -15,27 +15,19 @@
  *    See the License for the specific language governing permissions and
  *    limitations under the License.
  */
-
-/**
- *    @file
- *      This file defines object for a CHIP IM Invoke Command Handler
- *
- */
-
-#include "CommandHandler.h"
-#include "InteractionModelEngine.h"
-#include "RequiredPrivilege.h"
-#include "messaging/ExchangeContext.h"
+#include <app/CommandHandler.h>
```

Comment:
Would it be possible to keep header files associated with particular cpp sources with "local" includes, i.e. using " and without directory prefix? I'm asking because I'm not sure how well such convention (all includes as global includes) will work with include-what-you-use tool (which I'm using to track unnecessary/missing includes). I will check that that locally, though, but not before Monday.

Reply:
I considered it, however the trouble I had is that we have a lot of files in app and they seem to need to be in different "modules" rather than one large app one (e.g. some splittable things are the icd client & server, however others should be split to). 

The usage of includes was already very inconsistent, the only one consistently using relative imports were `src/app/MessageDef`. 

I could maybe make the CPP file include the same-named header as "local" however I am not sure we should. Are there specific errors in IWYU? Could you share how to run that one to try it out? I would expect full paths to make it easier for that tool rather than harder. As it stands I am not convinced our includes are sane - I had to fix at least one include that relied on include order and did not have `uint16_t` defined.

Reply:
Generally I would like to explicitly have full paths even for cpp files since these may reside in different source_sets.

I would however like to use IWYU to trim down things if possible - our includes are very odd right now (just yesterday we fixed a circular one and I believe we have a few more)

Reply:
Found we have some bits: #23092  ... will take a look and see how it works.

Reply:
I added a doc containing info about how to run IWYU. 

The tool does seem to work, except it proposes includes with '"' to be added instead of `<'. This seems fine though and probably usable.

I will do a future pass to fix up includes for src/app...

Reply:
Yes, some time ago we've added `iwyu-check.py` to ease some trouble with running that tool manually. I'm using it on platform code mostly, but there are some issues with it. First of all it is not consistent with "" and <> includes. Sometimes it expects "local" and sometimes <global> for the same file it expects <global> include. There are mapping added to https://github.com/project-chip/connectedhomeip/blob/master/scripts/helpers/platforms/iwyu.imp to fix some of these, but in the current form the mapping for "platform/internal/BLEManager.h" renders circular dependency in platform code (I'm not sure how to fix it, though...). 

Once I thought about adding this tool to CI, but I'm not sure whether that's a good idea and first of all, all includes would have to be fixed beforehand (which is not trivial sometimes).